base/linalg/triangular.jl

```diff
@@ -283,6 +281,90 @@ function -(A::UnitUpperTriangular)
     UpperTriangular(Anew)
 end
 
+# copy and scale
+function copy!{T<:Union{UpperTriangular, UnitUpperTriangular}}(A::T, B::T)
+    n = size(B,1)
+    for j = 1:n
+        for i = 1:(isa(B, UnitUpperTriangular)?j-1:j)
+            @inbounds A[i,j] = B[i,j]
+        end
+    end
+end
+function copy!{T<:Union{LowerTriangular, UnitLowerTriangular}}(A::T, B::T)
+    n = size(B,1)
+    for j = 1:n
+        for i = (isa(B, UnitLowerTriangular)?j+1:j):n
+            @inbounds A[i,j] = B[i,j]
+        end
+    end
+end
+function copy{T<:AbstractTriangular}(A::T)
+    D = similar(A)
+    copy!(D, A)
+    return D
+end
+for t in (:LowerTriangular, :UnitLowerTriangular, :UpperTriangular, :UnitUpperTriangular)
```

Comment:
Not sure here. I think the method you're referring to is
```javascript
copy(a::AbstractArray) = copy!(similar(a), a)
```
but because of the problem with `similar` and `BigFloat`I'd expect it to produce an half `#undef` matrix. I'll just check.